src/Microsoft.AspNetCore.HttpsPolicy/HstsMiddleware.cs

```diff
@@ -19,35 +20,39 @@ public class HstsMiddleware
     {
         private const string IncludeSubDomains = "; includeSubDomains";
         private const string Preload = "; preload";
+        private static readonly string[] _localhostStrings =
+        {
+            "localhost",
+            "127.0.0.1", // ipv4
+            "[::1]" // ipv6
+        };
 
         private readonly RequestDelegate _next;
         private readonly StringValues _strictTransportSecurityValue;
-
+        private readonly List<string> _excludedDomains;
         /// <summary>
         /// Initialize the HSTS middleware.
         /// </summary>
         /// <param name="next"></param>
         /// <param name="options"></param>
         public HstsMiddleware(RequestDelegate next, IOptions<HstsOptions> options)
         {
-            if (next == null)
-            {
-                throw new ArgumentNullException(nameof(next));
-            }
-
             if (options == null)
             {
                 throw new ArgumentNullException(nameof(options));
             }
 
-            _next = next;
+            _next = next ?? throw new ArgumentNullException(nameof(next));
 
             var hstsOptions = options.Value;
             var maxAge = Convert.ToInt64(Math.Floor(hstsOptions.MaxAge.TotalSeconds))
-                        .ToString(CultureInfo.InvariantCulture);
+                            .ToString(CultureInfo.InvariantCulture);
             var includeSubdomains = hstsOptions.IncludeSubDomains ? IncludeSubDomains : StringSegment.Empty;
             var preload = hstsOptions.Preload ? Preload : StringSegment.Empty;
             _strictTransportSecurityValue = new StringValues($"max-age={maxAge}{includeSubdomains}{preload}");
+
+            _excludedDomains = hstsOptions.AddHstsHeaderToLocahostRequests ? new List<string>() : new List<string>(_localhostStrings);
```

Comment:
Don't make this an option. Make it a default. If the user wants to send HSTS on localhost or add an additional domain to the list of excluded ones they can do it through options directly